docs/logging-usage.rst

```diff
@@ -0,0 +1,295 @@
+Using the API
+=============
+
+Authentication / Configuration
+------------------------------
+
+- Use :class:`Client <gcloud.logging.client.Client>` objects to configure
+  your applications.
+
+- :class:`Client <gcloud.logging.client.Client>` objects hold both a ``project``
+  and an authenticated connection to the Logging service.
+
+- The authentication credentials can be implicitly determined from the
+  environment or directly via
+  :meth:`from_service_account_json <gcloud.logging.client.Client.from_service_account_json>`
+  and
+  :meth:`from_service_account_p12 <gcloud.logging.client.Client.from_service_account_p12>`.
+
+- After setting ``GOOGLE_APPLICATION_CREDENTIALS`` and ``GCLOUD_PROJECT``
+  environment variables, create a :class:`Client <gcloud.logging.client.Client>`
+
+  .. doctest::
+
+     >>> from gcloud import logging
+     >>> client = logging.Client()
+
+Writing log entries
+-------------------
+
+Write a simple text entry to a log.
+
+.. doctest::
+
+   >>> from gcloud import logging
+   >>> client = logging.Client()
+   >>> log = client.log('log_name')
+   >>> log.text("A simple entry")  # API call
+
+Write a dictionary entry to a log.
+
+.. doctest::
+
+   >>> from gcloud import logging
+   >>> client = logging.Client()
+   >>> log = client.log('log_name')
+   >>> log.struct(message="My second entry",
+   ...     weather="partly cloudy")  # API call
+
+Retrieving log entries
+----------------------
+
+Fetch entries for the default project.
+
+.. doctest::
+
+   >>> from gcloud import logging
+   >>> client = logging.Client()
+   >>> entries, token = client.list_entries()  # API call
```

Comment:
We got feedback from the service teams that "implicit" service calls were a Bad Thing (TM).

We could probably write a generic iterator factory which would wrap around the "paged" methods, taking extra arguments as `**kw` to the factory.  E.g.:

```python
from gcloud.iterator import paging_iterator

for item in paging_iterator(client.list_entries, filter=FILTER, order_by=Client.DESCENDING):
    frob(item)
```

As long as the return values are the same, i.e., `(page_items, next_token)`, and the argument name for passing the previous token back in is consistent, it should Just Work (TM).